src/main/java/com/powsybl/integrationtest/creation/security/statemonitors/StateMonitorsSupplier.java

```diff
@@ -18,13 +18,15 @@
  *
  * @author Théo Le Colleter <theo.le-colleter at artelys.com>
  */
-public interface StateMonitorsProvider {
+public interface StateMonitorsSupplier {
 
     /**
      * Create and return a list of StateMonitor with every element that can be monitored (branches, voltage levels, three windings transformers).
      * Based on a list of contingencies, since at the creation of a StateMonitor, a contingency id must be given.
      * @param network, list of contingencies
      * @return a list of StateMonitor
      */
-    List<StateMonitor> createStateMonitorList(Network network, List<Contingency> contingencies);
+    List<StateMonitor> getStateMonitors(Network network, List<Contingency> contingencies);
+
+    public void setConfiguration(Object configuration);
```

Comment:
Same as for the `ContingenciesSupplier` : put the configuration directly in the `getStateMonitors` arguments.